doc/UsersGuide/source/ContributingDevelopment.rst

```diff
@@ -13,4 +13,284 @@ Contributing Development
+
+      # When the development is mature, tests have been conducted, and the developer is satisfied with the results, create a pull request to commit the code changes.
+
+      * Merge developer’s branch to the latest ufs-weather-model develop branch in authoritative repository. If changes are made in model sub-components, developers need to merge their branches to branches with the corresponding authoritative repository (or original repository for some components). For this, code management practices of the subcomponents need to be followed.
+
+      * Regression tests need to pass on at least one supported platform.
+
+      * For each component branch where developers make changes, developers need to go to their personal fork on GitHub and click on the “New pull request” button. When a new page “Compare changes” appears, developers will choose the branch in their fork with code changes to commit and the branch in upstream repository that the changes will be committed to. The code differences between the two branches will be displayed. Developers can review the differences and click on “submit pull request” to make the pull request. After code changes are committed to the component repository, developers will make pull requests to ufs-weather-model repository.
+
+It is suggested that the developers inform all the related code managers as the hierarchy structure of the ufs-weather-model repository may require collaboration among the code managers.
+
+-----------------------------------
+Engaging in the code review process
+-----------------------------------
+
+When code managers receive a pull request to commit the code changes, it is recommended that they add at least two code reviewers to review the code. The reviewers will write comments about the code changes and give a recommendation as to whether the code changes can be committed. What kinds of code changes will be accepted in the repository is beyond the scope of this document; future ufs-weather-model code management may have detailed answer for that.
+
+Reviewers may suggest some code changes during the review process. Developers need to respond to these comments in order to get code changes committed. If developers make further changes to their branch, reviewers need to check the code changes again. When both reviewers give recommendation to commit the code, code managers will merge the changes into the repository.
+
+---------------------------
+Conducting regression tests
+---------------------------
+
+Only developers that are running on a limited set of platforms (Hera, Cheyenne, WCOSS) can compile and run regression tests using the ufs-weather-model.
+
+To run regression test using rt.sh
+
+.. code-block:: console
+
+   % cd ufs-weather-model/tests
+   % ./rt.sh -f
+
+Regression test log files (ufs-weather-model/tests/Compile_$(MACHINE_ID).log and ufs-weather-model/tests/RegressionTests_$(MACHINE_ID).log ) will be updated.
+
```

Comment:
Provide explanation for when/why developer needs to create new baseline. Use content from Section 9.2.3.2 of CCPP TechDoc.

Reply:
9.2.3.2. Baselines
Regression testing is only possible on machines for which baselines exist, that is Tier 1 and 2 platforms.

When porting the code to a new machine, it is useful to start by establishing a personal baseline. Future runs of the RT can then be compared against the personal baseline to ascertain that the results have not been inadvertently affected by code developments. The rt.sh -c option is used to create a personal baseline.

./rt.sh -l rt.conf -c fv3 # create own reg. test baseline
Once the personal baseline has been created, future runs of the RT should be compared against the personal baseline using the -m option.

./rt.sh -l rt.conf -m # compare against own baseline